hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveCommits.java

```diff
@@ -66,4 +75,220 @@ public void testSuppressUnlockExceptions() throws TException, InterruptedExcepti
     // the commit must succeed
     Assert.assertEquals(1, ops.current().schema().columns().size());
   }
+
+  @Test
+  public void testThriftExceptionFailureOnCommit() throws TException, InterruptedException {
```

Comment:
Pretends we throw an error while persisting that actually fails to commit serverside

Reply:
nit: These onliner descriptions could be very useful as a comment in the code too so next generations could understand better the goal of these test.